tools/modules/initialize/message_checkpoint.sh

```diff
@@ -0,0 +1,67 @@
+
+module_options+=(
+	["set_checkpoint,author"]="@armbian"
+	["set_checkpoint,maintainer"]="@igorpecovnik"
+	["set_checkpoint,feature"]="set_checkpoint"
+	["set_checkpoint,example"]="help start mark stop show"
+	["set_checkpoint,desc"]="Helper module for timing code execution"
+	["set_checkpoint,status"]=""
+	["set_checkpoint,doc_link"]=""
+	["set_checkpoint,group"]="Development"
+	["set_checkpoint,port"]=""
+	["set_checkpoint,arch"]="x86-64 arm64"
+)
+#
+# Function to manage timer with multiple checkpoints
+function set_checkpoint() {
+	case "$1" in
+		help)
+		echo "Usage: set_checkpoint <start|stop|mark|show> [description] [show]"
+		echo "Commands:"
+		echo "  start              Start the timer."
+		echo "  stop               Stop the timer."
+		echo "  mark [description] [time] Mark a checkpoint with an optional description and an optional flag to show the output."
+		echo "  show               Show the total elapsed time and checkpoints."
+		;;
+		start)
+		set_checkpoint_START=$(date +%s)
+		set_checkpoint_CHECKPOINTS=()
+		set_checkpoint_DESCRIPTIONS=()
+		set_checkpoint_PREV=$set_checkpoint_START
+		;;
+		stop)
+		set_checkpoint_STOP=$(date +%s)
+		;;
+		mark)
+		local checkpoint_time=$(date +%s)
+		local checkpoint_duration=$((checkpoint_time - set_checkpoint_PREV))
+		set_checkpoint_PREV=$checkpoint_time
+		set_checkpoint_CHECKPOINTS+=($checkpoint_time)
+		set_checkpoint_DESCRIPTIONS+=("$2")
+		local count=${#set_checkpoint_DESCRIPTIONS[@]}
+		if [[ "$3" == "true" && $UXMODE == "true" ]]; then
+			printf "%-30s %10d seconds\n" "$2:" "${checkpoint_duration}"
+		fi
```

Comment:
Nitpick, but if you do this, you can avoid unnecessary calculations in non-UX mode:
```bash
	mark)
		if [[ "$UXMODE" == "true" ]]; then
			local checkpoint_time=$(date +%s)
			local checkpoint_duration=$((checkpoint_time - set_checkpoint_PREV))
			set_checkpoint_PREV=$checkpoint_time
			set_checkpoint_CHECKPOINTS+=($checkpoint_time)
			set_checkpoint_DESCRIPTIONS+=("$2")
			local count=${#set_checkpoint_DESCRIPTIONS[@]}
			[[ "$3" == "true" ]] && printf "%-30s %10d seconds\n" "$2:" "${checkpoint_duration}"
		fi
	;;
```